src/dotnet/APIView/APIViewWeb/Managers/APIRevisionsManager.cs

```diff
@@ -239,6 +243,37 @@ public async Task<bool> ToggleAPIRevisionApprovalAsync(ClaimsPrincipal user, str
             return updateReview;
         }
 
+
+        /// <summary>
+        /// Create APIRevision from File or FilePAth
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="review"></param>
+        /// <param name="file"></param>
+        /// <param name="filePath"></param>
+        /// <param name="language"></param>
+        /// <param name="label"></param>
+        /// <returns></returns>
+        public async Task<APIRevisionListItemModel> CreateAPIRevisionAsync(ClaimsPrincipal user, ReviewListItemModel review, IFormFile file, string filePath, string language, string label)
+        {
+            APIRevisionListItemModel apiRevision = null;
+
+            if (file != null)
+            {
+                using (var openReadStream = file.OpenReadStream())
+                {
+                    apiRevision = await AddAPIRevisionAsync(user: user, review: review, apiRevisionType: APIRevisionType.Manual,
+                        name: file.FileName, label: label, fileStream: openReadStream, language: language);
+                }
+            }
+            else if (!string.IsNullOrEmpty(filePath))
+            {
+                apiRevision = await AddAPIRevisionAsync(user: user, review: review, apiRevisionType: APIRevisionType.Manual,
+                           name: file.FileName, label: label, fileStream: null, language: language);
```

Comment:
`file` will be null when call comes here. Should this use `filePath` instead of `file.FileName`

Reply:
Thanks for catching that.